src/components/common/controllers/drag.ts

```diff
@@ -28,6 +34,13 @@ type DragState = State & {
   element: Element | null;
 };
 
+type Direction = {
+  isHorizontalMove: boolean;
+  movingDown: boolean;
+  movingToEndHorizontally: boolean;
+  movingToStartHorizontally: boolean;
+};
```

Comment:
Can this be a union type? Something along the lines of `left` | `top` | 'top-left`?